### ISSUE TITLE
JSON log timestamp format : milliseconds missing when 0

### ISSUE THREAD

**Issue by squalou** [keycloak/keycloak#29534]
### Description

The log format in use with json format is hard to parse - at least for some parsers, see details - because milliseconds are ignorer when zero.
The enhancement would be to include ".000" after the second in order to be parsed more easily.

For instance today I get this : 

`2024-05-14T16:21:56Z`

and I would like `2024-05-14T16:21:56.000Z`

### Motivation

I'm currently using fluentbit as a log parser / router to logging facility, which uses strptime format. (https://linux.die.net/man/3/strptime)  and so I use this format `%Y-%m-%dT%H:%M:%S.%LZ` ... which will fail when `.%L` is not there

I have sadly no other option (containers and cloud providers and so on : I have to stuck with fluentbit)

To be noted : I did have the exact same issue with SonarQube, and they nicely accepted to fix it, which gives me hope that the demand is not completely insane : )

**Comment by shawkins** [comment]
It looks like the default java formater has 0 as the min width for fractional seconds.

Try using the quakus property
[quarkus.log.console.json.date-format](https://quarkus.io/guides/logging#quarkus-logging-json_quarkus-log-console-json-date-format)=yyyy-MM-dd'T'HH:mm:ss.SSS'Z'

Or use the same file option, if you are using that logger instead.

cc @mabartos to assess if adding first-class option to keycloak would be worth while.

@squalou you could also open a quarkus issue to see if they will change the behavior of the default formatter.
